packages/rum-core/test/benchmarks/index.js

```diff
@@ -25,19 +25,22 @@
 
 import Transaction from '../../src/performance-monitoring/transaction'
 
+function getRandomInRange(min, max) {
+  return Math.random() * (max - min) + min
+}
+
 export function generateTestTransaction(noOfSpans, sampled) {
-  const tr = new Transaction(
-    `transaction with ${noOfSpans} spans`,
-    'transaction1type',
-    { startTime: 0, transactionSampleRate: sampled ? 1 : 0 }
-  )
-  let lastSpanEnd = 0
+  const tr = new Transaction(`transaction with ${noOfSpans} spans`, 'custom', {
+    startTime: 0,
+    transactionSampleRate: sampled ? 1 : 0
+  })
   for (let i = 0; i < noOfSpans; i++) {
-    const span = tr.startSpan(`span ${i}`, `span${i}type`, { startTime: 0 })
-    lastSpanEnd = 10 + i
-    span.end(lastSpanEnd)
+    const span = tr.startSpan(`name`, `type`, {
+      startTime: getRandomInRange(0, i)
+    })
+    span.end(getRandomInRange(i, i + 1))
   }
 
-  tr.end(lastSpanEnd + 100)
+  tr.end(noOfSpans + 10)
```

Comment:
This doesn't seem correct to me. end method should be called with a correct time, not with number of spans

Reply:
The logic is correct. `end` methods just needs a number so we can control the duration in our way and it would look like this 
```
const tr = generateTestTransaction(2)
tr.duration() // 12

// 2 created spans would be within this time range
```

Reply:
the syntax is correct but logically it doesn't make sense to bound the end of the transactions to the number of spans. Actually I think the previous logic is simpler and works just fine without having to introducing randomness into this. Is there any reason you chose this algorithm?

Reply:
I needed randomness while testing the grouping logic `groupSmallContinuouslySimilarSpans`. The previous logic required a intermediate variable `lastSpanEnd` which can be eliminated since its simpler to use `noOfSpans` as lastSpanEnd is within the range since our span.end makes sure its within the time range. 

Either way its not super important, we can simply throw in a fixed number and end the transaction.

Reply:
@vigneshshanmugam , Would you please elaborate on why randomness is need for testing `groupSmallContinuouslySimilarSpans`. If that's necessary I think we should at least add a comment here to explain why. At the moment it looks more like a mistake that we're using `noOfSpans + 10` as the end time of the transaction.

Reply:
I am happy to hear if you have any ideas here. The randomness was required so we can group few spans lave rest as it is and to keep them differentiated. In any case, I am happy to tune the method if looks complicated.

Reply:
We have discussed offline and agreed on removing the randomness and create a fixed number of groupable/ungroupable spans.